store/tikv/snapshot.go

```diff
@@ -277,6 +277,11 @@ func (s *tikvSnapshot) Get(ctx context.Context, k kv.Key) ([]byte, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	err = s.store.CheckVisibility(s.version.Ver)
```

Comment:
Why not `CheckVisibility` before `get`, so does the others, e.g. batch get, cop task?

Reply:
1. Still need `CheckVisibility ` after `get`.  But we can do double-check too. Because If only check before `get`, the gc safe point maybe changed after check, so currently, all `CheckVisibility` check was after `get`.
2. Already add tests for others.

Reply:
I guess @jackysp means move this check before line 275